go-controller/pkg/ovn/gress_policy.go

```diff
@@ -168,36 +238,40 @@ func (gp *gressPolicy) getMatchFromIPBlock(lportMatch, l4Match string) string {
 }
 
 // addNamespaceAddressSet adds a new namespace address set to the gress policy
+
 func (gp *gressPolicy) addNamespaceAddressSet(name, portGroupName string) {
-	hashName := hashedAddressSet(name)
-	if gp.nsAddressSets.Has(hashName) {
+	v4HashName := hashedAddressSet(getIPv4ASName(name))
+	v6HashName := hashedAddressSet(getIPv6ASName(name))
+
+	if gp.peerAddressSets[ipv4].Has(v4HashName) || gp.peerAddressSets[ipv6].Has(v6HashName) {
```

Comment:
I still think || makes sense for the add scenario, after making the code changes as in case of single stack, only one of them will be populated. Is there any scenarios where && needs to be considered?